app/models/model.rb

```diff
@@ -18,4 +18,22 @@ class Model < ApplicationRecord
   scoped_search :on => :info
   scoped_search :on => :vendor_class, :complete_value => :true
   scoped_search :on => :hardware_model, :complete_value => :true
+
+  def hosts_count
+    Host::Managed.authorized.group(:model_id).count.fetch(id, 0)
```

Comment:
Will this create a lot of n+1 queries?

Reply:
I will look into it. You might be right.

Reply:
I believe this means that there is no `N+1`:

```ruby
irb(main):119:0> Host::Managed.authorized.group(:model_id).count
D, [2018-07-02T16:57:23.009902 #4328] DEBUG -- :    (0.2ms)  SELECT COUNT(*) AS count_all, "hosts"."model_id" AS hosts_model_id FROM "hosts" WHERE "hosts"."type" IN ('Host::Managed') GROUP BY "hosts"."model_id"
=> {1=>11, 2=>1}
```